### PR TITLE
fix(hmr): replacement of optionalChain is compatible with legacy versions

### PR DESCRIPTION
**Pull request by wChenonly**
fix: #7495 

## Checklist

- [x] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

**Review by JSerFeng — APPROVED** [review]

File: packages/rspack/src/builtin-plugin/css-extract/hmr/hotModuleReplacement.ts [hunk 1]
```diff
@@ -50,19 +50,16 @@ function getCurrentScriptUrl(moduleId: string) {
...
-	return (fileMap: string): Option<Array<string>> => {
+	return (fileMap: string): Option<Array<string>> | null => {
 		if (!src) {
 			return null;
 		}
 
-		const splitResult = src.split(/([^\\/]+)\.js$/);
-		const filename = splitResult?.[1];
+		const splitResult = src.match(/([^\\/]+)\.js$/);
+		// biome-ignore lint/complexity/useOptionalChain: <explanation>
```
Comment (JSerFeng) [inline packages/rspack/src/builtin-plugin/css-extract/hmr/hotModuleReplacement.ts:59]:
```suggestion
		// biome-ignore lint/complexity/useOptionalChain: not use optionalChain to support legacy browser
```

File: packages/rspack/src/builtin-plugin/css-extract/hmr/hotModuleReplacement.ts [hunk 2]
```diff
@@ -227,7 +234,8 @@ function cssReload(moduleId: string, options: Record<string, any>) {
 		}
 
 		if (reloaded) {
-			console.log("[HMR] css reload %s", src?.join(" "));
+			// biome-ignore lint/complexity/useOptionalChain: <explanation>
```
Comment (JSerFeng) [inline packages/rspack/src/builtin-plugin/css-extract/hmr/hotModuleReplacement.ts:237]:
```suggestion
			// biome-ignore lint/complexity/useOptionalChain: not use optionalChain to support legacy browser
```

File: tests/plugin-test/css-extract/cases/hmr-locals/expected/main.js [hunk 3]
```diff
@@ -40,12 +40,10 @@ function getCurrentScriptUrl(moduleId) {
...
             return null;
         }
-        const splitResult = src.split(/([^\\/]+)\.js$/);
-        const filename = splitResult?.[1];
-        if (!filename) {
-            return [src.replace(".js", ".css")];
-        }
-        if (!fileMap) {
+        const splitResult = src.match(/([^\\/]+)\.js$/);
+        // biome-ignore lint/complexity/useOptionalChain: <explanation>
```
Comment (JSerFeng) [inline tests/plugin-test/css-extract/cases/hmr-locals/expected/main.js:44]:
```suggestion
        // biome-ignore lint/complexity/useOptionalChain: not use optionalChain to support legacy browser
```

File: tests/plugin-test/css-extract/cases/hmr-locals/expected/main.js [hunk 4]
```diff
@@ -167,7 +173,8 @@ function cssReload(moduleId, options) {
             return;
         }
         if (reloaded) {
-            console.log("[HMR] css reload %s", src?.join(" "));
+            // biome-ignore lint/complexity/useOptionalChain: <explanation>
```
Comment (JSerFeng) [inline tests/plugin-test/css-extract/cases/hmr-locals/expected/main.js:176]:
```suggestion
            // biome-ignore lint/complexity/useOptionalChain: not use optionalChain to support legacy browser
```

File: tests/plugin-test/css-extract/cases/hmr/expected/main.js [hunk 5]
```diff
@@ -40,12 +40,10 @@ function getCurrentScriptUrl(moduleId) {
...
             return null;
         }
-        const splitResult = src.split(/([^\\/]+)\.js$/);
-        const filename = splitResult?.[1];
-        if (!filename) {
-            return [src.replace(".js", ".css")];
-        }
-        if (!fileMap) {
+        const splitResult = src.match(/([^\\/]+)\.js$/);
+        // biome-ignore lint/complexity/useOptionalChain: <explanation>
```
Comment (JSerFeng) [inline tests/plugin-test/css-extract/cases/hmr/expected/main.js:44]:
```suggestion
        // biome-ignore lint/complexity/useOptionalChain: not use optionalChain to support legacy browser
```

File: tests/plugin-test/css-extract/cases/hmr/expected/main.js [hunk 6]
```diff
@@ -167,7 +173,8 @@ function cssReload(moduleId, options) {
             return;
         }
         if (reloaded) {
-            console.log("[HMR] css reload %s", src?.join(" "));
+            // biome-ignore lint/complexity/useOptionalChain: <explanation>
```
Comment (JSerFeng) [inline tests/plugin-test/css-extract/cases/hmr/expected/main.js:176]:
```suggestion
            // biome-ignore lint/complexity/useOptionalChain: not use optionalChain to support legacy browser
```

**Comment by JSerFeng** [comment]
Thanks ! 💙

**Comment by wChenonly** [comment]
> Thanks ! 💙

✅ done

**Review by JSerFeng — APPROVED** [review]

**Pull request merged**